build/efr32mg12/efr32-openthread.mk

```diff
@@ -0,0 +1,286 @@
+# the EFR32 file to ensure that OpenThread is configured properly for the
+# for EFR32 platforms.
+OPENTHREAD_PROJECT_CONFIG = openthread-core-$(OPENTHREAD_TARGET)-config.h
+
+# Additional header files needed by the EFR32 port of OpenThread
+# but not installed automatically by OpenThread's build system.
+OPENTHREAD_PLATFORM_HEADERS = \
+    $(OPENTHREAD_ROOT)/examples/platforms/$(OPENTHREAD_TARGET)/openthread-core-efr32-config.h \
+    $(OPENTHREAD_ROOT)/examples/platforms/openthread-system.h
+
+# ==================================================
+# Compilation flags / settings specific to building
+#   OpenThread itself.
+# ==================================================
+
+OPENTHREAD_CPPFLAGS = $(STD_CFLAGS) $(DEBUG_FLAGS) $(OPT_FLAGS) -Wno-expansion-to-defined \
+                      $(OPENTHREAD_DEFINE_FLAGS) $(OPENTHREAD_INC_FLAGS)
+                      
+OPENTHREAD_CXXFLAGS = $(STD_CXXFLAGS) -Wno-expansion-to-defined -fno-exceptions -fno-exceptions -fno-rtti 
+OPENTHREAD_DEFINE_FLAGS = $(foreach def,$(OPENTHREAD_DEFINES),-D$(def))
+OPENTHREAD_INC_FLAGS = $(foreach dir,$(OPENTHREAD_INC_DIRS),-I$(dir))
+
+OPENTHREAD_LDFLAGS              = \
+    -specs=nosys.specs            \
+    -Wl,--gc-sections
+
+OPENTHREAD_DEFINES = \
+    EFR32 \
+    OPENTHREAD_PROJECT_CORE_CONFIG_FILE='\"$(OPENTHREAD_PROJECT_CONFIG)\"' \
+    MBEDTLS_CONFIG_FILE='\"mbedtls-config.h\"' \
+    MBEDTLS_USER_CONFIG_FILE='\"efr32mg12-weave-mbedtls-config.h\"' \
+    RADIO_CONFIG_DMP_SUPPORT=1 \
+    $(MCU) 
+    
+OPENTHREAD_INC_DIRS = \
+	  $(PROJECT_ROOT) \
+  	$(OPENWEAVE_ROOT)/src/adaptations/device-layer/include/Weave/DeviceLayer/EFR32/crypto \
+    $(OPENTHREAD_ROOT)/examples/platforms \
+    $(OPENTHREAD_ROOT)/examples/platforms/$(OPENTHREAD_TARGET) \
+    $(OPENTHREAD_ROOT)/examples/platforms/$(OPENTHREAD_TARGET)/crypto \
+    $(OPENTHREAD_ROOT)/third_party/mbedtls \
+    $(OPENTHREAD_ROOT)/third_party/mbedtls/repo/include \
+    $(OPENTHREAD_ROOT)/third_party/mbedtls/repo/include/mbedtls \
+    $(EFR32_SDK_ROOT)/platform/CMSIS/Include \
+    $(EFR32_SDK_ROOT)/platform/Device/SiliconLabs/EFR32MG12P/Include \
+    $(EFR32_SDK_ROOT)/platform/emlib/inc \
+    $(EFR32_SDK_ROOT)/platform/radio/rail_lib/chip/efr32/efr32xg1x \
+    $(EFR32_SDK_ROOT)/platform/radio/rail_lib/protocol/ble \
+    $(EFR32_SDK_ROOT)/platform/radio/rail_lib/common \
+    $(EFR32_SDK_ROOT)/util/third_party/mbedtls/configs \
+    $(EFR32_SDK_ROOT)/util/third_party/mbedtls/sl_crypto/include \
+    $(EFR32_SDK_ROOT)/protocol/bluetooth/ble_stack/inc/soc \
+    $(EFR32_SDK_ROOT)/protocol/bluetooth/ble_stack/inc/common \
+    $(EFR32_SDK_ROOT)/app/bluetooth/common/util \
+    $(HAL_CONF_DIR)
+
+CONFIG_FILE      = OPENTHREAD_PROJECT_CORE_CONFIG_FILE='\"openthread-core-efr32-config.h\"'
+CONFIG_FILE_PATH = $(OPENTHREAD_ROOT)/examples/platforms/$(OPENTHREAD_TARGET)/
+HAL_CONF_DIR     = $(CONFIG_FILE_PATH)/$(shell echo $(BOARD) | tr A-Z a-z)
```

Comment:
This will need modification of OpenThread as its makefiles depend on the BOARD variabl from within the platforms/efr32xx/brdxxxx directories.  We currently have 3 directories in OpenThread for MG12, MG21 and now MG13 families.  As a lot of code is common we intend to merge these into a single common directory.  As part of this we can look to add support for custom boards and associated hal config files.